src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,791 @@
+                        {
+                            var methodSymbol = semanticModel.GetDeclaredSymbol(method, cancellationToken);
+                            if (methodSymbol is null)
+                            {
+                                modifiedMembers.Add(method);
+                                continue;
+                            }
+
+                            if (methodSymbol.Name == "Clone")
+                            {
+                                // delete any 'clone' method as it is reserved in records
+                                continue;
+                            }
+
+                            // TODO: Ensure the single param is a StringBuilder
+                            if (methodSymbol.Name == "PrintMembers" &&
+                                methodSymbol.ReturnType.SpecialType == SpecialType.System_Boolean &&
+                                methodSymbol.Parameters.Length == 1)
+                            {
+                                if (type.TypeKind == TypeKind.Class && !type.IsSealed)
+                                {
+                                    // ensure virtual and protected modifiers
+                                    // TODO: change from virtual to override if inheriting from a record
+                                    modifiedMembers.Add(method.WithModifiers(SyntaxFactory.TokenList(
+                                        SyntaxFactory.Token(SyntaxKind.ProtectedKeyword),
+                                        SyntaxFactory.Token(SyntaxKind.VirtualKeyword))));
+                                }
+                                else
+                                {
+                                    // ensure private member
+                                    modifiedMembers.Add(method.WithModifiers(
+                                        SyntaxFactory.TokenList(SyntaxFactory.Token(SyntaxKind.PrivateKeyword))));
+                                }
+
+                                continue;
+                            }
+
+                            if (methodSymbol.Name == nameof(GetHashCode) &&
+                                methodSymbol.ReturnType.SpecialType == SpecialType.System_Int32 &&
+                                methodSymbol.Parameters.IsEmpty &&
+                                UseSystemHashCode.Analyzer.TryGetAnalyzer(semanticModel.Compilation, out var analyzer))
+                            {
+                                var methodOperation = semanticModel.GetOperation(method, cancellationToken);
+                                if (methodSymbol != null && methodOperation is IMethodBodyOperation methodBodyOperation)
+                                {
+                                    var (accessesBase, members, statements) = analyzer.GetHashedMembers(methodSymbol, methodOperation);
+                                    // TODO: check to see if the properties are the same as we would use and delete
```

Comment:
so, imo, i would far prefer to not have all thse todos in teh PR.  if it's not done yet, just remove, and we can add in further followup PRs.